### PR TITLE
Fetched iOS device and used in iOS test

### PR DESCRIPTION
**Pull request by siarhei-luskanau**

### CODE REVIEW

File: src/jvmTest/kotlin/wizard/GeneratedProjectTest.kt [hunk 1]
```diff
@@ -144,6 +144,20 @@ class GeneratedProjectTest {
...
         )
         val dir = projectInfo.writeToDir(workingDir)
+        val deviceId = checkCommand(
+            dir = dir,
+            command = listOf(
+                "xcrun",
+                "simctl",
+                "list",
+                "devices",
+                "available"
```
Comment (terrakok) [inline src/jvmTest/kotlin/wizard/GeneratedProjectTest.kt:154]: it is better to use `xcrun simctl list devices available -j` and parse a JSON output

Comment (siarhei-luskanau) [reply src/jvmTest/kotlin/wizard/GeneratedProjectTest.kt]: JSON output is applied.

File: src/jvmTest/kotlin/wizard/GeneratedComposeAppProjectTest.kt [hunk 2]
```diff
@@ -164,8 +192,8 @@ class GeneratedComposeAppProjectTest {
                 "Debug",
                 "OBJROOT=${dir.path}/build/ios",
                 "SYMROOT=${dir.path}/build/ios",
-                "-sdk",
-                "iphonesimulator",
```
Comment (siarhei-luskanau) [inline src/jvmTest/kotlin/wizard/GeneratedComposeAppProjectTest.kt:168]: "-sdk iphonesimulator" doesn't work for some reasons if I launch tests on my local machine. But "-destination id=..." works well locally and on github actions.

File: build.gradle.kts [hunk 3]
```diff
@@ -32,6 +32,7 @@ kotlin {
         }
         commonTest.dependencies {
             implementation(kotlin("test"))
+            implementation("com.google.code.gson:gson:2.10.1")
```
[the remark this answers is not in the document]
Comment (terrakok) [reply build.gradle.kts:35]: I don't get you. Just apply the gradle plugin and add the dependency.
https://github.com/Kotlin/kotlinx.serialization?tab=readme-ov-file#gradle

[the remark this answers is not in the document]
Comment (siarhei-luskanau) [reply build.gradle.kts:35]: let me try...

File: src/jvmTest/kotlin/wizard/GeneratedComposeAppProjectTest.kt [hunk 4]
```diff
@@ -163,6 +166,32 @@ class GeneratedComposeAppProjectTest {
...
+                .filter { it.isAvailable }
+                .filter {
+                    if (true.toString().equals(other = System.getenv("CI"), ignoreCase = true)) {
+                        listOf("iphone 15", "iphone 14").any { device -> it.name.contains(device, true) }
+                    } else {
+                        true
+                    }
+                }
+            }
+            .values.flatten()
```
Comment (terrakok) [inline src/jvmTest/kotlin/wizard/GeneratedComposeAppProjectTest.kt:192]: BTW, are you sure we need a static types here?
We can use regular java JSON class to work with the devicesJson as with plain string-to string map

Comment (siarhei-luskanau) [reply src/jvmTest/kotlin/wizard/GeneratedComposeAppProjectTest.kt]: Implemented using the `JSONObject`

File: build.gradle.kts [hunk 5]
```diff
@@ -32,6 +32,7 @@ kotlin {
         }
         commonTest.dependencies {
             implementation(kotlin("test"))
+            implementation("org.json:json:20240303")
```
Comment (terrakok) [inline build.gradle.kts:35]: I meant `implementation("javax.json:javax.json-api:1.1.4")` 🤔

File: src/jvmTest/kotlin/wizard/GeneratedComposeAppProjectTest.kt [hunk 6]
```diff
@@ -163,6 +164,34 @@ class GeneratedComposeAppProjectTest {
...
+            )
+        )
+        val devicesList = Json.createReader(StringReader(devicesJson)).use { it.readObject() }
+            .getJsonObject("devices")
+            .let { devicesMap -> devicesMap.keys.map { devicesMap.getJsonArray(it) } }
+            .map { jsonArray -> jsonArray.map { it.asJsonObject() } }
+            .flatten()
+            .filter { it.getBoolean("isAvailable") }
+            .filter {
+                if (true.toString().equals(other = System.getenv("CI"), ignoreCase = true)) {
```
Comment (terrakok) [inline src/jvmTest/kotlin/wizard/GeneratedComposeAppProjectTest.kt:185]: is it the same as `System.getenv("CI").toBoolean()` ?

Comment (siarhei-luskanau) [reply src/jvmTest/kotlin/wizard/GeneratedComposeAppProjectTest.kt]: It looks better

**Pull request merged**